### PR TITLE
RELATED: RAIL-1815 - Support execution by reference

### PR DESCRIPTION
**Pull request by lupko**
Updated the `forInsightByRef` signature & contract; implemented it for bear & added `executeByReference` flag to InsightView.

In the end, doing InsightView with the flag on and on top of bear will lead to use of `executeVisualization` endpoint with added filters & resultSpec.

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/724/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2312/ : SUCCESS in 11m 11s

File: libs/sdk-backend-base/src/toolkit/execution.ts [hunk 1]
```diff
@@ -55,4 +62,57 @@ export abstract class AbstractExecutionFactory implements IExecutionFactory {
...
         return this.forDefinition(def);
     }
+
+    public forInsightByRef(insight: IInsight, filters?: IFilter[]): IPreparedExecution {
+        return this.forInsight(insight, filters);
+    }
+}
+
+/**
+ * This implementation of execution factory allows transparent injection of fixed set of filters to all
```
Comment (no23reason) [inline libs/sdk-backend-base/src/toolkit/execution.ts:72]: Maybe we should mention that this is a concat and not the custom logic we use in InsightView for example.

File: libs/sdk-backend-bear/src/backend/workspace/execution/preparedExecutionByRef.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        return this._fingerprint;
+    }
+
+    public equals(other: IPreparedExecution): boolean {
+        return isEqual(this.definition, other.definition);
+    }
+}
+
+function checkDefIsExecutable(_def: IExecutionDefinition): void {
+    return;
```
Comment (no23reason) [inline libs/sdk-backend-bear/src/backend/workspace/execution/preparedExecutionByRef.ts:93]: What should this do?

File: libs/sdk-backend-bear/src/backend/workspace/execution/test/executionFactory.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { BearAuthenticatedCallGuard } from "../../../../types/auth";
+import { NotSupported } from "@gooddata/sdk-backend-spi";
+import { ReferenceLdm, ReferenceRecordings } from "@gooddata/reference-workspace";
+import { recordedInsight } from "@gooddata/sdk-backend-mockingbird";
+import { newAttributeSort } from "@gooddata/sdk-model";
+import { withCaching } from "@gooddata/sdk-backend-base";
+import bearFactory from "../../../../index";
+
+const UnsupportedAuthCall: BearAuthenticatedCallGuard = () => {
+    throw new NotSupported("you got too far");
```
Comment (no23reason) [inline libs/sdk-backend-bear/src/backend/workspace/execution/test/executionFactory.test.ts:13]: That is quite a cryptic message, can we make it more helpful?

File: libs/sdk-backend-bear/src/backend/workspace/execution/preparedExecutionByRef.ts (hunk 2 above)
Comment (lupko) [reply libs/sdk-backend-bear/src/backend/workspace/execution/preparedExecutionByRef.ts:93]: dunno, it was in the freeform execution and got carried through. i'll delete from both

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/317/ : FAILURE in 2m 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/573/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2321/ : SUCCESS in 17m 28s

**Pull request merged**